src/Modules/UngradedModules/FreeResolutions.jl

```diff
@@ -791,9 +799,9 @@ julia> length(fr)
 """
 function length(F::FreeResolution)
   if !is_complete(F.C)
-    idx = findfirst(is_zero(domain(phi)) for phi in F.C.maps)
+    idx = findfirst(is_zero(F.C[i]) for i = 0:first(map_range(F.C)))
```

Comment:
The above code runs into an error in Julia 1.6 (so another reason to drop support for that, @benlorenz @lgoettgens). But this is a workaround (albeit also a de-optimization, but hopefully a minor one, since most resolutions probably won't have millions of steps...)
```suggestion
    idx = findfirst([is_zero(F.C[i]) for i = 0:first(map_range(F.C))])
```

Reply:
```suggestion
    idx = findfirst(i -> is_zero(F.C[i]), 0:first(map_range(F.C)))
```
This is also a workaround, but without a performance downgrade